src/engine/strat_engine/thinpool/thinpool.rs

```diff
@@ -259,6 +259,17 @@ fn status_to_usage(status: Option<ThinPoolStatus>) -> Option<ThinPoolUsage> {
     })
 }
 
+/// Convert the thin pool status to the metadata low water mark.
```

Comment:
This is fine (exactly what we do for ThinPoolUsage), but I'm starting to wonder why we don't just store the whole ThinPoolStatus in the thinpool implementation and extract the data we need from that when we want it. We can either tackle that in this PR, or I'll file an issue and we can work on that later.

Reply:
Resolved.